File: Day3/part1.py
# Get lines from file
with open ("Day3/data.txt", "r") as file:

    lines = file.read().splitlines()

total = 0
prior_chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"

for line in lines:

    # Split each line in half
    left, right = line[:len(line) // 2], line[len(line) // 2:]

    # Check for matching chars in both halves
    match_char = ""
    
    for char in left:

        if char in right:

            match_char = char
            break

    # Get priority value relative to matching char
    for i, char in enumerate(prior_chars):

        if match_char == char:

            total += i + 1

            print(f"{match_char} = {i + 1}")

print(total)
    

File: Day3/part1.c
#include <stdio.h>
#include <string.h>
#include "data.h"

int main(int argc, char const *argv[])
{
    
    int i;
    int j;
    int k;
    int left_index_start;
    int right_index_start;
    int left_index_end;
    int right_index_end;
    int half;

    for (i = 0; i < 300; i++)
    {

        half = strlen(data[i]) / 2;

        char left[50];
        char right[50];

        for (j = 0; j < half; j++)
        {
            
            left[j] = data[i][j];

        }

        left[j] = '\0';

        for(j = half, k = 0; j <= strlen(data[i]); j++, k++) {
        
            right[k]= data[i][j];
 
        }

        printf("%s | %s", left, right);
        
        
        printf("\n\n");

    }

    return 0;

}
